bin/gh-md-toc

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env bash
+
+#
+# Steps:
+#
+#  1. Download corresponding html file for some README.md:
+#       curl -s $1
+#
+#  2. Discard rows where no substring 'user-content-' (github's markup):
+#       awk '/user-content-/ { ...
+#
+#  3.1 Get last number in each row like ' ... </span></a>sitemap.js</h1'.
+#      It's a level of the current header:
+#       substr($0, length($0), 1)
+#
+#  3.2 Get level from 3.1 and insert corresponding number of spaces before '*':
+#       sprintf("%*s", substr($0, length($0), 1)*3, " ")
+#
+#  4. Find head's text and insert it inside "* [ ... ]":
+#       substr($0, match($0, /a>.*<\/h/)+2, RLENGTH-5)
+#
+#  5. Find anchor and insert it inside "(...)":
+#       substr($0, match($0, "href=\"[^\"]+?\" ")+6, RLENGTH-8)
+#
+
+gh_toc_version="0.6.1"
+
+gh_user_agent="gh-md-toc v$gh_toc_version"
+
+#
+# Download rendered into html README.md by its url.
+#
+#
+gh_toc_load() {
+    local gh_url=$1
+
+    if type curl &>/dev/null; then
```

Comment:
I usually use `command -v`.
https://stackoverflow.com/a/677212/885053

For me in zsh
```console
$ type wget
wget not found
$ type wget &>/dev/null
$
```

Also, while you set the shebang, I've had the experience of seeing people execute the script with a different tool (eg `sh`), and bash-specific stuff ended breaking.

Maybe this is a small detail, but that could have become an issue on which we'd have spent a while trying to figure out the issue if we didn't already know about it ^^

Reply:
The link doesn't really have a explanation on how command -v really behaves. What would be the exact change? `if  command -v &> /dev/null; then`?

It mostly doesn't matter for us as it should soon always run in the same environment that we control (through GH actions), but it might be worth putting upstream to https://github.com/ekalinin/github-markdown-toc

Reply:
Let's forget about it. If we have an issue with that later, we know of to resolve it (just execute the script as is, or with bash). But yeah, if you want to post on that project, feel free to.
with command, I'd do something like
```bash
if [[ -z "$(command -v wget)" ]]; then
fi
```